src/Sarif.Driver/Sdk/AnalyzeCommandBase.cs

```diff
@@ -601,18 +603,38 @@ private SupportedPlatform GetCurrentRunningOS()
                     context.TargetUri.GetFileName()));
             }
 
-            if (options.ComputeFileHashes && _resultsCachingLogger.HashToResultsMap.TryGetValue(context.Hashes.Sha256, out List<Tuple<ReportingDescriptor, Result>> results))
+            if (options.ComputeFileHashes)
             {
-                context.Logger.AnalyzingTarget(context);
-                foreach (Tuple<ReportingDescriptor, Result> result in results)
+                _resultsCachingLogger.HashToResultsMap.TryGetValue(context.Hashes.Sha256, out List<Tuple<ReportingDescriptor, Result>> results);
+                _resultsCachingLogger.HashToNotificationsMap.TryGetValue(context.Hashes.Sha256, out List<Notification> notifications);
+
+                bool replayCachedData = (results != null || notifications != null);
+
+                if (replayCachedData)
```

Comment:
>replayCachedData [](start = 20, length = 16)

This logic ensures that if we've already seen this file, we don't repeat the analysis. We just log the existing results for this file. And `ResultCachingLogger` ensures that repeated results appear only once in the results list. So far so good. But why even call the logger if we've already seen this file? Won't that result in getting the same console output repeatedly for the file, even if we don't get duplicate results? IOW, why _do_ we "replay the cached data". Wasn't once enough?

Reply:
Yes, see my comment below. I misunderstood the scenario. We are encountering the same file as a dependency of different analysis targets (not multiple times in the course of analyzing a single target), so the URIs will be different.

---
In reply to: [316889393](https://github.com/microsoft/sarif-sdk/pull/1658#discussion_r316889393) [](ancestors = 316889393)